bluesky/callbacks/core.py

```diff
@@ -3,15 +3,55 @@
 """
 from itertools import count
 import warnings
+import os
+
 from collections import deque, namedtuple, OrderedDict
 import time as ttime
+from functools import wraps as _wraps, partial as _partial
 
 from datetime import datetime
 import logging
 from ..utils import ensure_uid
+from event_model import DocumentNames
+
+
+logger = logging.getLogger(__name__)
+
+
+def make_callback_safe(func=None, *, logger=None):
+    if func is None:
+        return _partial(make_callback_safe, logger=logger)
+
+    @_wraps(func)
+    def inner(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except Exception as ex:
+            if logger is not None:
+                logger.exception(ex)
```

Comment:
I believe this expects a string message (suggestion: “This exception occurred in a ‘safe’ callback and will be ignored to avoid interrupting acquisition.”) The exception is detected automatically from context and need not be passed in.